src/matcalc/_surface.py

```diff
@@ -0,0 +1,303 @@
+      3. Optionally relax each slab and evaluate its total energy.
+      4. Compute the surface energy from the difference between slab and bulk
+         (per-atom) energies, divided by the slab surface area. For a symmetric
+         slab, it is assumed there are 2 identical surfaces.
+
+    Typical usage:
+      1. Instantiate with all relevant parameters (e.g. ``miller_index``,
+         ``min_slab_size``, ``relax_bulk``).
+      2. Call :meth:`calc_slabs` on a Pymatgen bulk :class:`Structure`.
+         This returns a dictionary of slab structures and internally stores
+         bulk data.
+      3. Pass that slab dictionary to :meth:`calc` to get a final dictionary
+         of surface energies for each slab.
+      4. Optionally, use :meth:`calc_many` to process multiple inputs in parallel.
+
+    Example:
+        >> my_calc = SomeASECalculator(...)
+        >> surf_calc = SurfaceCalc(calculator=my_calc, miller_index=(1,0,0), ...)
+        >> slabs_dict = surf_calc.calc_slabs(bulk_structure)
+        >> results = surf_calc.calc(slabs_dict)
+        >> print(results)
+
+    :ivar calculator: The ASE Calculator object used for energy/force evaluations.
+    :type calculator: Calculator
+    :ivar miller_index: Miller index for the surface, e.g. (1, 0, 0).
+    :type miller_index: tuple[int,int,int]
+    :ivar min_slab_size: Thickness of the slab in Å.
+    :type min_slab_size: float
+    :ivar min_vacuum_size: Thickness of the vacuum region in Å.
+    :type min_vacuum_size: float
+    :ivar relax_bulk: Whether to relax the bulk structure (cell+atomic DOFs).
+    :type relax_bulk: bool
+    :ivar relax_slab: Whether to relax each slab structure (atomic DOFs).
+    :type relax_slab: bool
+    :ivar fmax: Force tolerance (eV/Å) for relaxation.
+    :type fmax: float
+    :ivar optimizer: The ASE optimizer to use, e.g. "BFGS" or an :class:`Optimizer`.
+    :type optimizer: str | Optimizer
+    :ivar max_steps: Maximum number of optimization steps.
+    :type max_steps: int
+    :ivar relax_calc_kwargs: Additional keyword arguments passed to :class:`RelaxCalc`.
+    :type relax_calc_kwargs: dict | None
+    """
+
+    def __init__(
+        self,
+        calculator: Calculator,
+        *,
+        miller_index: tuple[int, int, int] = (1, 0, 0),
+        min_slab_size: float = 10.0,
+        min_vacuum_size: float = 20.0,
+        relax_bulk: bool = True,
+        relax_slab: bool = True,
+        fmax: float = 0.1,
+        optimizer: str | Optimizer = "BFGS",
+        max_steps: int = 500,
+        relax_calc_kwargs: dict | None = None,
+    ) -> None:
+        """
+        Constructor for initializing the SurfaceCalc with all parameters needed
+        to generate and optionally relax bulk and slab structures.
+
+        :param calculator: ASE Calculator for energy/force evaluations.
+        :type calculator: Calculator
+        :param miller_index: Miller index for the slab surfaces (e.g., (1, 0, 0)).
```

Comment:
I don;t think the miller-index should be part of the init. After, the calc method can take a slab of any miller-index.

Reply:
Same for min and max slab sizes.